PostHog/PostHogSDK.swift

```diff
@@ -44,7 +44,8 @@ let maxRetryDelay = 30.0
     private var flagCallReported = Set<String>()
     private var featureFlags: PostHogFeatureFlags?
     private var context: PostHogContext?
-    private static var apiKeys = Set<String>()
+    private static let instancesLock = NSLock()
+    private static var instances: [String: Weak<PostHogSDK>] = [:]
```

Comment:
i have not done that on android, wondering if this is a better (or not) approach? can we chat about this?

Reply:
discussed offline, we don't need this.